### PR TITLE
Pay/libplugin fixes, ccan modules for JSON handling.

### LINKED ISSUE #2612: 'pay' plugin is killed and ceases to exist when called with wrong parameters
i don't know if this is the expected behavior. If it is, please close this issue.

I called the `pay` command with msatoshi = `null`, got the response `{ "code" : -32601, "message" : "Unknown command 'pay'" }`. Tried again, got the same response.

Checked the logs:
`lightningd[24617]: 2019-05-01T12:42:52.214Z plugin-pay Killing plugin: Failed to parse JSON response '??{ "jsonrpc": "2.0", "id": 24,  "error" :  { "code" : -32602, "message" : ""msatoshi" should be a millisatoshi amount, not "null"" } }??'`

Then I tried to call it again with correct parameters, but now it was too late, `pay` was gone.

`listconfigs` before showed a `"plugin": "/home/fiatjaf/lightning/lightningd/../plugins/pay"`, now it doesn't have that key.

### PR DESCRIPTION
**Pull request by rustyrussell**
I moved various JSON things into ccan, so I could reuse it for libplugin.  The cleanups along the way
broke pylightning compat, so I disabled that, but if you prefer to have it for longer I can hack the JSON
back again.

Fixes #2612

### CODE REVIEW

**Comment by rustyrussell** [comment]
Trivial rebase

File: plugins/libplugin.c [hunk 1]
```diff
@@ -619,21 +649,23 @@ struct plugin_timer *plugin_timer(struct plugin_conn *rpc, struct timerel t,
...
+	json_out_addstr(jout, "level",
+			l == LOG_DBG ? "debug"
+			: l == LOG_INFORM ? "info"
+			: l == LOG_UNUSUAL ? "warn"
+			: "error");
+	json_out_addv(jout, "message", true, fmt, ap);
+	json_out_end(jout, '}');
+
+	/* Last '}' is done by finish_and_send_json */
+	finish_and_send_json(STDOUT_FILENO, jout);
```
Comment (cdecker) [inline plugins/libplugin.c]: ~Still strange that we don't use the higher level helpers for `json_stream` here, they'd also check the JSON stream is well formed.~

Nevermind, it migrates to `json_out` in later commits anyway :-)

File: tests/test_pay.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 
 def test_pay(node_factory):
-    l1, l2 = node_factory.line_graph(2)
+    l1, l2 = node_factory.line_graph(2, opts={'log-level': 'io'})
```
Comment (cdecker) [inline tests/test_pay.py]: Leftover from testing?

File: tests/test_gossip.py [hunk 3]
```diff
@@ -752,7 +752,7 @@ def fund_from_to_payer(lsrc, ldst, lpayer):
 
     # Setup
     # Construct lightningd
-    l1, l2, l3, l4 = node_factory.get_nodes(4)
+    l1, l2, l3, l4 = node_factory.get_nodes(4, opts={'log_all_io': True})
```
Comment (cdecker) [inline tests/test_gossip.py]: Leftover from testing?

**Comment by rustyrussell** [comment]
Rebased to fix trivial CHANGELOG conflict, and remove debugging code @cdecker found...

**Pull request merged**